A/A doc cleanup, A/A ops doc, index names adjusted (#1657)
hashicorp/terraform-website: content/source/docs/enterprise/install/active-active.html.md

As mentioned above, the Active/Active architecture requires an existing [automated installation](https://www.terraform.io/docs/enterprise/install/automating-the-installer.html) of Terraform Enterprise that follows our [best practices for deployment](https://www.terraform.io/docs/enterprise/before-installing/reference-architecture/index.html). 

> The primary requirement is an auto scaling group (or equivalent) with a single instance running Terraform Enterprise. This ASG should be behind a load balancer and can be exposed to the public Internet or not depending on your requirements. As mentioned earlier, the installation of the Terraform Enterprise application should be automated completely so that the ASG can be scaled to zero and back to one without human intervention. 

Reviewer: Use of the abbreviated ASG without defining it here.

Author: corrected

The application itself must be using [External Services](https://www.terraform.io/docs/enterprise/before-installing/index.html#operational-mode-decision) mode to connect to an external PostgreSQL database and object storage. 

> All admin and application configuration must be automated via your settings files and current with running configuration, i.e. it cannot have been altered via the Admin UI and not synced to the file. Specifically, you should be using the following configuration files:

Reviewer 2: Admin UI is referred to as Replicated Admin Console in the admin document. I think we should start to standardize on a single term for this concept.

Author: Changed all occurrences. Also clarified a reference to just UI to Terraform Enterprise UI.

*   `/etc/replicated.conf`    - contains the configuration for the Replicated installer
*   `/etc/ptfe-settings.json` - contains the configuration for the TFE application 

-> **Note**: The location for the latter is controlled by the "ImportSettingsFrom" setting in `/etc/replicated.conf` and is sometimes named `settings.json` or `replicated-tfe.conf`

The requirement for automation is two-fold. First, the nodes need to be able to spin up and down without human intervention. More importantly though, you will need to ensure configuration is managed in this way going forward as the Admin UI will be disabled. **The Admin UI does not function correctly when running multiple nodes and must not be used**.

### Step 1: Prepare to Externalize Redis

#### Prepare Network

There are new access requirements involving ingress and egress:

> *   **Port 6379** (or the port the external Redis will be configured to use) must be open between the node instances and the Redis service
> *   **Port 8201** must be open between the node instances to allow Vault to run in [High Availability](https://www.vaultproject.io/docs/internals/high-availability) mode

Reviewer 2: ```suggestion
*   **Port 6379** (or the port the external Redis will be configured to use) must be open between the nodes and the Redis service
*   **Port 8201** must be open between the nodes to allow Vault to run in [High Availability](https://www.vaultproject.io/docs/internals/high-availability) mode
```

Author: I had actually standardized on node instance, but missed some. I debated just node, but wasn't sure we wanted to depict too much of a true "cluster", so went node instance. I think just node cleaner and since you prepared the commits already, will accept all those revisions.

> *   **Port 8800** should now be closed, as the admin dashboard is no longer available when running multiple nodes

Reviewer 2: Admin dashboard is yet an other term for the Replicated console 😁

Author: also changed

> The primary requirement is an Auto Scaling Group (or equivalent) with a single instance running Terraform Enterprise. This ASG should be behind a Load Balancer and can be exposed to the public internet or not depending on your requirements. As mentioned earlier, the installation of the Terraform Enterprise application should be automated completely so that the ASG can be scaled to zero and back to one without human intervention. 

Reviewer 3: ```suggestion
The primary requirement is an auto scaling group (or equivalent) with a single instance running Terraform Enterprise. This ASG should be behind a load balancer and can be exposed to the public Internet or not depending on your requirements. As mentioned earlier, the installation of the Terraform Enterprise application should be automated completely so that the ASG can be scaled to zero and back to one without human intervention. 
```

> The application itself must be using [External Services](https://www.terraform.io/docs/enterprise/before-installing/index.html#operational-mode-decision) mode to connect to an external PostgreSQL database and blob storage. 

Reviewer 3: ```suggestion
The application itself must be using [External Services](https://www.terraform.io/docs/enterprise/before-installing/index.html#operational-mode-decision) mode to connect to an external PostgreSQL database and object storage. 
```

### Step 2: Update your Configuration File Templates

Before installing, you need to make changes to the templates for the configuration files mentioned earlier in the prerequisites. 

#### Update Installer Settings

> The existing settings for the installer and infrastructure (`replicated.conf`) are still needed and require to be expanded. Please see documentation for the existing options [here](https://www.terraform.io/docs/enterprise/install/automating-the-installer.html#installer-settings). 

Reviewer 4: ```suggestion
The existing settings for the installer and infrastructure (`replicated.conf`) are still needed and require to be expanded. Please see documentation for the existing options [here](/docs/enterprise/install/automating-the-installer.html#installer-settings). 
```

#### Update Application Settings

> The existing settings for the Terraform Enterprise application (`ptfe-settings.json`) are still needed and require to be expanded. Please see documentation for the existing options [here](https://www.terraform.io/docs/enterprise/install/automating-the-installer.html#application-settings).

Reviewer 4: ```suggestion
The existing settings for the Terraform Enterprise application (`ptfe-settings.json`) are still needed and require to be expanded. Please see documentation for the existing options [here](/docs/enterprise/install/automating-the-installer.html#application-settings).
```

   
   "user_token":{
      "value":"6c403112981d0ae2ce6b4b2c7a1b1997"

>    },

Reviewer 2: The trailing comma after the "registry_session_secret_key" should be removed, per the JSON specification. Can't add this comment to that line, though 🤷🏻

Author: And you do syntax checking! corrected.

```

> -> **Note**: The individual settings need to match across node instances. The values should not be the same, e.g. `cookie_hash` and `internal_api_token` would not contain identical values.

Reviewer 2: ```suggestion
-> **Note**: The individual settings need to match across nodes. The values should not be the same, e.g. `cookie_hash` and `internal_api_token` would not contain identical values.
```

### Step 3: Connect to External Redis

Once you are prepared to include the modified configuration options in your configuration files, you must connect a single node to your newly provisioned Redis service by rebuilding your node instance with the new settings.

#### Re-provision Terraform Enterprise Instance

Terminate the existing instance by scaling down to zero. Once terminated, you can scale back up to one instance using your revised configuration. 

#### Wait for Terraform Enterprise to Install

> It can take up to 15 minutes for the node instance to provision and the TFE application to be installed and respond as healthy. You can monitor the status of the node instance provisioning by watching your Auto Scaling Group in your cloud’s web console. To confirm the successful implementation of the TFE application you can SSH onto the instance and run the following command to monitor the installation directly:

Reviewer 2: ```suggestion
It can take up to 15 minutes for the node to provision and the TFE application to be installed and respond as healthy. You can monitor the status of the node instance provisioning by watching your Auto Scaling Group in your cloud’s web console. To confirm the successful implementation of the TFE application you can SSH onto the instance and run the following command to monitor the installation directly:
```

Installation is complete once `isTransitioning` is `false` and `State` is `started`.

> See [Active/Active Administration](https://www.terraform.io/docs/enterprise/admin/active-active.html) for more status and troubleshooting commands.

Reviewer 4: ```suggestion
See [Active/Active Administration](/docs/enterprise/admin/active-active.html) for more status and troubleshooting commands.
```

Installation is complete once `isTransitioning` is `false` and `State` is `started`.

See [Active/Active Administration](https://www.terraform.io/docs/enterprise/admin/active-active.html) for more status and troubleshooting commands.

#### Validate Application

> With installation complete, it is time to validate the new Redis connection. Terraform Enterprise uses Redis both as a cache for API requests and a queue for long running jobs, e.g. Terraform Runs.  Test the latter behavior by running real Terraform Plans & Applies through the system. 

Reviewer 3: ```suggestion
With installation complete, it is time to validate the new Redis connection. Terraform Enterprise uses Redis both as a cache for API requests and a queue for long running jobs, e.g. Terraform Runs. Test the latter behavior by running real Terraform plans and applies through the system. 
```

> #### Disable the Admin Dashboard
> 
> Before scaling beyond the first node, you must disable the  as mentioned earlier in this guide. This is done by adding the `disable-replicated-ui` flag as a parameter when you call the install script, as such:

Reviewer 2: Looks like the "admin dashboard" phrase just got dropped here. Also, the title of this section should be adjusted for consistency.

```suggestion
#### Disable the Replicated Admin Console

Before scaling beyond the first node, you must disable the Replicated Admin Console as mentioned earlier in this guide. This is done by adding the `disable-replicated-ui` flag as a parameter when you call the install script, as such:
```

Author: I think the one edit got hosed in my last commit on this topic. The section title was missed. Both corrected. (couldn't use yoour suggested commit as due to the hosing, it thinks there are deleted lines and doesn't support that??)

Now that you have tested your external Redis connection change the min and max instance count of your Auto Scaling Group to two nodes. 

> -> **Note**: Running more than two instances is not _currently_ supported. Please contact your Technical Account Manager if there are any questions/concerns.

Reviewer 3: ```suggestion
-> **Note**: Running more than two instances is not _currently_ supported. Please contact your Customer Success Manager if there are any questions/concerns.
```

#### Wait for Terraform Enterprise to Install

You need to wait up to 15 minutes for the application to respond as healthy on both nodes. Monitor the status of the install with the same methods used previously for one node in Step 3. 

-> **Note**: Each node needs to be checked independently.

#### Validate Application

> Finally, confirm the application is functioning as expected when running multiple nodes. Run Terraform Plan and Applies through the system (and any other tests specific to your environment) like you did to validate the application in Step 3. 

Reviewer 3: ```suggestion
Finally, confirm the application is functioning as expected when running multiple nodes. Run Terraform plan and applies through the system (and any other tests specific to your environment) like you did to validate the application in Step 3. 
```

The default example provided on the provider page can be used to deploy memorystore [here](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/redis_instance). The host output of the resource can then be provided to the terraform module in order to configure connectivity.

> You may consider using other options in the configuration depending on your requirements, such as including the **auth_enabled** flag set to true, which must then be accompanied by including an additional tfe configuration item called **redis_password** set to the value returned in the **auth_string** attribute from the memorystore resource.  

Reviewer 3: ```suggestion
You may consider using other options in the configuration depending on your requirements, such as including the **auth_enabled** flag set to true, which must then be accompanied by including an additional TFE configuration item called **redis_password** set to the value returned in the **auth_string** attribute from the memorystore resource.  
```
